### ISSUE TITLE
[Feature] enable NAT traversal for simple 1-level NAT situation with UPNP support: connecting local host validator behind a NAT to public leaders

### ISSUE THREAD

**Issue by jz0000** [harmony-one/harmony#2959]
# Problem/limitation At Hand

This is a derived feature of issue 2865 "support libp2p autonat",  which intends to add full NAT support for p2p host. This issue here is the sub-issue or 1st step to solve the problem: enable NAT traversal in a simple 1-level NAT situation where a validator node behind a NAT (with UPNP support) is trying to connect to leaders in public IP. 

**Which component?**
- [x] p2p
- [ ] consensus
- [ ] deploy script
- [ ] wallet
- [ ] explorer
- [ ] node script
- [ ] test automation
- [ ] demo app
- [ ] others ...

# Proposed Solution

The nice thing about this first step is that it only needs to add "NatPortMap" option when starting a new Host and this already covers most of situations in our current harmony network: leaders almost always have public IPs and only validators could be behind NAT. 

It will configure an outer-ip/port  to  inner-ip/port mapping in the NAT as the host starts to talk to outside world, it does this by calling any upnp tools available on the NAT.  

# Additional Context

the libp2p-autonat library can support more general use cases of NAT traversal by setting up relay nodes in the public IPs and let them relay packets between nodes behind NATs, which seems not so urgent given current harmony network operation. We can add that support as we go along.

**Comment by jz0000** [comment]
Some details about the testbed I set up in a VPC in AWS:

I have set up a virtual private cloud (VPC) on AWS with 2 subnets: the 1st one is a public subnet which can be accessed from outside; and the 2nd one is a private subnet with only internal private IP addresses. I launched two EC2 instances, one inside public subnet with a public IP and one inside private subnet with only private IP. Then I also launched a NAT instance with a public IP to connect the two subnets. Then in the private subnet I configured a route table where all the IPs destined to outside the private subnet point to this nat instance, where IP translation is happening: SIP will be changed to NAT's own IP for outgoing packets and DIP will be changed to local private IP for incoming packets. (Note that this external/internal mapping will be done automatically once we enable the NATPortMAP option in libp2p host, if any router has UPNP support on it)

And I build a native harmony node on the server inside private subnet and launched it, it is running smoothly with blocks synced correctly from the whole harmony network outside private subnet. This proves that once the NAT IP/port mapping is configured on the router or in this case a dedicated NAT instance, harmony node can run on it smoothly.
